cmd/lotus-wallet/interactive.go

```diff
@@ -0,0 +1,240 @@
+
+				if builtin.IsMultisigActor(toact.Code) && cmsg.Method == multisig.Methods.Propose {
+					var mp multisig.ProposeParams
+					if err := mp.UnmarshalCBOR(bytes.NewReader(cmsg.Params)); err != nil {
+						return xerrors.Errorf("unmarshalling multisig propose params: %w", err)
+					}
+
+					fmt.Println("\tMultiSig Proposal Value:", types.FIL(mp.Value))
+					fmt.Println("\tMultiSig Proposal Hex Params:", hex.EncodeToString(mp.Params))
+
+					toact, err := napi.StateGetActor(ctx, mp.To, types.EmptyTSK)
+					if err != nil {
+						return xerrors.Errorf("looking up msig dest actor: %w", err)
+					}
+
+					fmt.Println("\tMultiSig Proposal Method:", stmgr.MethodsMap[toact.Code][mp.Method].Name)
+					p, err := lcli.JsonParams(toact.Code, mp.Method, mp.Params)
+					if err != nil {
+						return err
+					}
+
+					fmt.Println("\tMultiSig Proposal Params:", strings.ReplaceAll(p, "\n", "\n\t"))
+				}
+			} else {
+				fmt.Println("Params: No chain node connection, can't decode params")
+			}
+
+		case api.MTDealProposal:
+			return xerrors.Errorf("TODO") // TODO
+		default:
+			log.Infow("WalletSign", "address", k, "type", meta.Type)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return c.under.WalletSign(ctx, k, msg, meta)
+}
+
+func (c *InteractiveWallet) WalletExport(ctx context.Context, a address.Address) (*types.KeyInfo, error) {
+	err := c.accept(func() error {
+		fmt.Println("-----")
+		fmt.Println("ACTION: WalletExport - Export private key")
+		fmt.Printf("ADDRESS: %s\n", a)
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return c.under.WalletExport(ctx, a)
+}
+
+func (c *InteractiveWallet) WalletImport(ctx context.Context, ki *types.KeyInfo) (address.Address, error) {
+	err := c.accept(func() error {
+		fmt.Println("-----")
+		fmt.Println("ACTION: WalletImport - Import private key")
+		fmt.Printf("TYPE: %s\n", ki.Type)
+		return nil
+	})
+	if err != nil {
+		return address.Undef, err
+	}
+
+	return c.under.WalletImport(ctx, ki)
+}
+
+func (c *InteractiveWallet) WalletDelete(ctx context.Context, addr address.Address) error {
+	err := c.accept(func() error {
+		fmt.Println("-----")
+		fmt.Println("ACTION: WalletDelete - Delete a private key")
+		fmt.Printf("ADDRESS: %s\n", addr)
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	return c.under.WalletDelete(ctx, addr)
+}
+
+func (c *InteractiveWallet) accept(prompt func() error) error {
+	c.lk.Lock()
+	defer c.lk.Unlock()
+
+	if err := prompt(); err != nil {
+		return err
+	}
+
+	yes := randomYes()
+	for {
+		fmt.Printf("\nAccept the above? (%s/No): ", yes)
+		var a string
+		if _, err := fmt.Scanln(&a); err != nil {
+			return err
+		}
+		switch a {
+		case yes:
+			fmt.Println("approved")
+			return nil
+		case "No":
+			return xerrors.Errorf("action rejected")
+		}
+
+		fmt.Printf("Type EXACTLY '%s' or 'No'\n", yes)
+	}
+}
+
+var yeses = []string{
+	"yes",
+	"Yes",
+	"YES",
+	"approve",
+	"Approve",
+	"sign",
```

Comment:
sign/Sign isn't a good "random yes" because you're also using these for the prompt in `WalletNew`